proposals/NNNN-Vehicle-data-refactoring.md

```diff
@@ -0,0 +1,598 @@
+	</history>
+</param>
+```
+
+##### Update function `GetVehicleData` response: 
+
+```xml
+<param name="prndl" type="PRNDL" mandatory="false" since="2.0" until="X.x">
+	<description>See PRNDL</description>
+	<history>
+		<param name="prndl" type="PRNDL" mandatory="false"/>
+	</history>
+</param>
+```
+
+##### Update function `OnVehicleData` notification: 
+
+```xml
+<param name="prndl" type="PRNDL" mandatory="false" since="2.0" until="X.x">
+	<description>See PRNDL</description>
+	<history>
+		<param name="prndl" type="PRNDL" mandatory="false"/>
+	</history>
+</param>
+```
+
+#### Updates in HMI_API: 
+
+
+##### Remove element `VEHICLEDATA_PRNDL` from enum `VehicleDataType` in `Common` interface: 
+
+```xml
+-<element name="VEHICLEDATA_PRNDL"/>
+```
+
+##### Remove enum `PRNDL` from `Common` interface:
+
+```xml
+-<enum name="PRNDL">
+-	<description>The selected gear.</description>
+-	<element name="PARK">
+-		<description>Parking</description>
+-	</element>
+	...
+- 	<element name="FAULT">
+- 	</element>
+-</enum>
+```
+
+
+##### Remove param `prndl` from function `SubscribeVehicleData` request: 
+
+```xml
+-<param name="prndl" type="Boolean" mandatory="false">
+- 	<description>See PRNDL</description>
+-</param>
+```
+
+##### Remove param `prndl` from function `SubscribeVehicleData` response: 
+
+```xml
+-<param name="prndl" type="Common.VehicleDataResult" mandatory="false">
+-	<description>See PRNDL</description>
+-</param>
+```
+
+##### Remove param `prndl` from function `UnsubscribeVehicleData` request: 
+
+```xml
+-<param name="prndl" type="Boolean" mandatory="false">
+- 	<description>See PRNDL</description>
+-</param>
+```
+
+##### Remove param `prndl` from function `UnsubscribeVehicleData` response: 
+
+```xml
+-<param name="prndl" type="Common.VehicleDataResult" mandatory="false">
+-	<description>See PRNDL</description>
+-</param>
+```
+
+
+##### Remove param `prndl` from function `GetVehicleData` request: 
+
+```xml
+-<param name="prndl" type="Boolean" mandatory="false">
+- 	<description>See PRNDL</description>
+-</param>
+```
+
+##### Remove param `prndl` from function `GetVehicleData` response: 
+
+```xml
+-<param name="prndl" type="Common.PRNDL" mandatory="false">
+-	<description>See PRNDL</description>	
+-</param>
+```
+
+##### Remove param `prndl` from function `OnVehicleData` response: 
+
+```xml
+-<param name="prndl" type="Common.PRNDL" mandatory="false">
+-	<description>See PRNDL</description>	
+-</param>
+```
+
+## Potential downsides
+
+Since this is a breaking change, HMI would need to be updated as well.
+ 
+## Impact on existing code
+
+* SDL Core needs to be updated as per new API.
+* Proxy needs to be updated to support getters/setters as per new API.
+* SDL Server needs to update permissions for refactored vehicle data items.
+* HMI needs to be updated to support refactored structures.
+
+## Alternatives considered
+
+* Instead of adding `hazardLights`  to `TurnSignal`, it can be a new vehicle data items in itself.
```

Comment:
```suggestion
* Instead of adding `hazardLights` to `TurnSignal`, it can be a new vehicle data item in itself.
```

Reply:
Done, i am going to move TurnSignal to new proposal.